maxbalves/PostBy
# Added Local Data Refresh + Like/Dislike Safety Checks

**Local Data Refresh:**
When posts are edited, liked, disliked, or deleted:
- Timeline automatically refreshes to display those changes through a delegate
- Any deleted posts are removed from:
    - the timeline and the tableView is reloaded (no Parse re-querying)
    - the map and the pins are recreated (no Parse re-querying)

**Like/Dislike Safety Checks**
When a post is liked or disliked:
- The original post is queried in the database by objectId
    - If not found, delegate pops the view controller and removes post from timeline or map (as talked before)
- If found, the post queried from the database is changed to have the same like/dislike relation to the user as the local post
- The local post is set to be the new queried post
- The data of the post and the user is saved to the database

_Tests For Like/Dislike Checks_:
As talked on commit, only basic tests with the same account have been done. For example:
- Like post from home page -> through maps, dislike post by going to details screen -> back in home page, unlike the post
- Result: the post should have no like or dislike since the local copy will overwrite the previous choice on the database. 

[TODO]
- Talk to Project Peer on Wednesday 3pm about possible solutions for Like/Dislike feature (due to slowness)
   - {possible semi-solution}:
       - change the local values and display them
       - use asynchronous calls to query and save updated post data
       - update the display again when the asynchronous calls are done
       - [conclusion of solution]: it will hide the slowness of saving things and make the user feel as if things are fast. _however_ there's a chance the user will see the like/dislike counter suddenly change after async calls are done
- For now, will start working on the comments feature.

**Overall Demo:**
<img src="http://g.recordit.co/FJ4Pc43D8t.gif" width=250px>
